web/packages/teleterm/src/services/appUpdater/clientToolsUpdateProvider.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { app } from 'electron';
+import {
+  AppUpdater,
+  DebUpdater,
+  MacUpdater,
+  NsisUpdater,
+  Provider,
+  ResolvedUpdateFileInfo,
+  RpmUpdater,
+  UpdateInfo,
+} from 'electron-updater';
+import { ProviderRuntimeOptions } from 'electron-updater/out/providers/Provider';
+
+const CHECKSUM_FETCH_TIMEOUT = 5_000;
+
+/** Implements electron-updater's `Provider` with client tools updates. */
+export class ClientToolsUpdateProvider extends Provider<UpdateInfo> {
+  constructor(
+    private getClientToolsVersion: ClientToolsVersionGetter,
+    private nativeUpdater: AppUpdater,
+    runtimeOptions: ProviderRuntimeOptions
+  ) {
+    super(runtimeOptions);
+  }
+
+  override async getLatestVersion(): Promise<UpdateInfo> {
+    const clientTools = await this.getClientToolsVersion();
+
+    // If no client tools version is specified, return the current version
+    // to simulate an up-to-date state.
+    if (!clientTools) {
+      return {
+        version: app.getVersion(),
+        releaseDate: '',
+        path: '',
+        sha512: '',
+        files: [],
+      };
+    }
+
+    const { baseUrl, version } = clientTools;
+    const fileUrl = `https://${baseUrl}/${makeDownloadFilename(this.nativeUpdater, version)}`;
+    const sha256 = await fetchChecksum(fileUrl);
```

Comment:
In what circumstances is this fetch going to get trigger? I'm thinking if 5s is enough of a timeout or if we should bump it to 10s in case the network is struggling. But I just don't know how often we'll call this.